pkg/api/norman/customization/cluster/formatter.go

```diff
@@ -61,16 +45,45 @@ func (f *Formatter) Formatter(request *types.APIContext, resource *types.RawReso
 	resource.Links["shell"] = shellLink
 	resource.AddAction(request, v32.ClusterActionGenerateKubeconfig)
 	resource.AddAction(request, v32.ClusterActionImportYaml)
+
+	// If user has permissions to update the cluster (regardless of RKE1 or not)
+	if canUpdateCluster(request) {
```

Comment:
The old implementation will pass `resource.Values` to the `CanDo` function. This implementation passes `map[string]interface{}{"id": request.ID}`.

Are we sure this doesn't introduce potential issues?

Reply:
Fixed and retested